worker.py

```diff
@@ -47,7 +47,10 @@ def detachVM(self, return_vm=False, replace_vm=False):
         or not in the pool (replace_vm). The worker must always call
         this function before returning.
         """
-        if return_vm:
+        # job-owned instance, simply destroy after job is completed
+        if self.job.accessKeyId:
```

Comment:
This if branch action is the same as the else statement. They both do `safeDestroyVM`. Any reason for this?